### PR TITLE
XIVY-12550 Add CatPath Chooser

### PR DESCRIPTION
**Pull request by ivy-edp**
Implemented the Category Path Chooser with mock tests
grafik

### CODE REVIEW

File: packages/editor/src/components/browser/categorie/CatPathChooser.tsx [hunk 1]
```diff
@@ -1,21 +1,157 @@
...
+    const location = props.location.toLowerCase();
+    switch (true) {
+      case location.includes('request') || location.includes('start'):
+        return 'START';
+      case location.includes('case'):
+        return 'CASE';
+      case location.includes('task'):
+        return 'TASK';
+      default:
+        throw new Error('Unknown type');
```
Comment (ivy-lli) [inline packages/editor/src/components/browser/categorie/CatPathChooser.tsx:48]: are you sure about this?
If this happens maybe the whole UI is broken. Is this what we want?
Or did you try it?
I think we can simply fire a request with an unknown location and the backend should then simply return nothing or something like this...

File: packages/editor/src/components/widgets/code-editor/useCodeEditor.ts [hunk 2]
```diff
@@ -38,8 +38,14 @@ export const useMonacoEditor = (options?: { modifyAction?: ModifyAction }) => {
...
     } else {
-      const text = value.cursorValue.length > 0 && options?.modifyAction ? options.modifyAction(value.cursorValue) : value.cursorValue;
+      const text =
+        value.cursorValue.length > 0 && options?.modifyAction && type !== 'catPath'
+          ? options.modifyAction(value.cursorValue)
+          : value.cursorValue;
       editor.executeEdits('browser', [{ range: selection, text, forceMoveMarkers: true }]);
+      if (type === 'catPath') {
+        editor.setValue(text);
+      }
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/code-editor/useCodeEditor.ts:48]: arr, a little bit ugly, but I think I understand why it is needed...
Maybe a sign that we should somehow change this or at last refactor this code a bit into smaller readable functions...

File: packages/protocol/src/data/inscription.ts [hunk 3]
```diff
@@ -6,7 +6,8 @@
  * and run json-schema-to-typescript to regenerate this file.
  */
 
-export type PID = string
+export type Type1 = ("START" | "TASK" | "CASE")
```
Comment (ivy-lli) [inline packages/protocol/src/data/inscription.ts:9]: ugly name... can't we improve this @ivy-rew?

File: packages/protocol/src/data/inscription.ts [hunk 4]
```diff
@@ -21,7 +22,7 @@ export type HttpMethod = "GET" | "POST" | "PUT" | "HEAD" | "DELETE" | "PATCH" |
 export type InputType = "ENTITY" | "FORM" | "RAW";
 export type WsAuth = "NONE" | "WS_SECURITY" | "HTTP_BASIC";
 export type Severity = "INFO" | "WARNING" | "ERROR";
-export type Type = "START" | "INTERMEDIATE" | "ACTIVITY";
+export type Type2 = "START" | "INTERMEDIATE" | "ACTIVITY";
```
Comment (ivy-lli) [inline packages/protocol/src/data/inscription.ts:25]: ugly name... can't we improve this @ivy-rew?

**Review by ivy-lli — COMMENTED** [review]
Looks good.
I would think about throwing an error -> this will not help any user I think.

And I'm not sure if the category browser should be a tree... Of course the macros will not be possible, but all other we could easily render as tree.
And I think I would not call this browser "Category Path chooser", only "Category".

File: packages/protocol/src/data/inscription.ts (hunk 4 above)
Comment (ivy-rew) [reply packages/protocol/src/data/inscription.ts:25]: yes indeed; we should change that type-info in the backend ... by renaming our new object to "CategoryType"

Comment (ivy-edp) [reply packages/protocol/src/data/inscription.ts:25]: yes, I can do that👍

File: packages/protocol/src/data/inscription.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ export type HttpMethod = "GET" | "POST" | "PUT" | "HEAD" | "DELETE" | "PATCH" |
 export type InputType = "ENTITY" | "FORM" | "RAW";
 export type WsAuth = "NONE" | "WS_SECURITY" | "HTTP_BASIC";
 export type Severity = "INFO" | "WARNING" | "ERROR";
-export type Type2 = "START" | "INTERMEDIATE" | "ACTIVITY";
+export type Type = "START" | "INTERMEDIATE" | "ACTIVITY";
```
Comment (ivy-lli) [inline packages/protocol/src/data/inscription.ts:25]: I think a better name for this would be nice too, but not important for now

**Review by ivy-lli — APPROVED** [review]
Technically good now I think. But as discussed IMO this should be a combobox like the signal boundary has.

**Pull request closed without merging**